### PR TITLE
NEON: Implement some f16XN types and f16 related intrinsics.

### PR DESCRIPTION
**Pull request by yyctw**
Hi all, this is Eric from Andes Technology Corporation. This PR includes
1. Add the types `simde_float16x4x2_t` and `simde_float16x8x2_t`
2. Add 105 initial implementations and corresponding test cases in 35 families which are listed below:
- `mul`, `sub`, `ext`, `div`, `min`, `max`, `fma`, `abs`, `cgt`, `clt`, `clez`, `padd`, `pmax`
- `get_lane`, `get_low`, `get_high`, `mul_lane`, `mul_n`, `set_lane`
- `combine`, `cvt`, `ld2`, `ld1_dup`, `st2`, `zip1`, `zip2`, `zip`, `reinterpret`
- `rsqrts`, `recps`, `recpe`, `rndn`, `rsqrte`, `sqrt`, `qrshrnh_n`

All CI tests have passed on my fork! Thanks for reading and any recommendations are welcome!

### CODE REVIEW

**Comment by mr-c** [comment]
@yyctw Thank you very much for this contribution! Can you rebase your branch to bring it up to date?

**Comment by yyctw** [comment]
Thank you for your suggestion! I've already rebased my branch to ensure it's up to date.

**Comment by yyctw** [comment]
> @yyctw Thank you very much for this contribution! Can you rebase your branch to bring it up to date?

File: simde/arm/neon/div.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Copyright:
+ *   2020      Evan Nemerson <evan@nemerson.com>
+ *   2020      Sean Maher <seanptmaher@gmail.com> (Copyright owned by Google, LLC)
```
Comment (mr-c) [inline simde/arm/neon/div.h:25]: Please replace this with your own copyright lines (and feel free to add yours to the other files you contributed to)

**Review by mr-c — CHANGES_REQUESTED** [review]
Looking great! One request about the copyright lines

File: simde/arm/neon/div.h (hunk 1 above)
Comment (yyctw) [reply simde/arm/neon/div.h:25]: Of course, I will promptly add the copyright lines to the files I contributed to. Thanks!

**Review by mr-c — APPROVED** [review]

**Pull request merged**